### PR TITLE
Add support for Whatsapp ID when sending contacts

### PR DESCRIPTION
**Pull request by vicenterusso**
Not sure about the `__construct` param orders though

### CODE REVIEW

**Comment by netflie** [comment]
> 

I don't like the param order but I think this option is the best option in order to don't break the API and don't need to release a major version of the package. This will be refactored in future major versions, not problem.

**Review by netflie — CHANGES_REQUESTED** [review]
Hi!

Thanks for your contributing! Unit tests are mandatory to integrate the PR ^^

File: src/Message/Contact/Phone.php [hunk 1]
```diff
@@ -6,11 +6,14 @@ class Phone
...
 
+    protected string $wa_id;
+
     protected PhoneType $type;
 
-    public function __construct(string $number, PhoneType $type)
+    public function __construct(string $number, PhoneType $type, string $wa_id = null)
     {
         $this->number = $number;
+        $this->wa_id = $wa_id ?? $number;
```
[the remark this answers is not in the document]
Comment (vicenterusso) [reply src/Message/Contact/Phone.php:16]: According to [documentation](https://developers.facebook.com/docs/whatsapp/on-premises/reference/contacts) I think we can safely say that Whatsapp User ID is in indeed the phone number.

Not sure about the nullable. Probably it's better not being set if not sharing a non whatsapp number

**Review by netflie — APPROVED** [review]

File: src/Message/Contact/Phone.php [hunk 2]
```diff
@@ -6,11 +6,14 @@ class Phone
...
 
+    protected string $wa_id;
+
     protected PhoneType $type;
 
-    public function __construct(string $number, PhoneType $type)
+    public function __construct(string $number, PhoneType $type, string $wa_id = null)
     {
         $this->number = $number;
+        $this->wa_id = $wa_id ?? '';
```
Comment (netflie) [inline src/Message/Contact/Phone.php:16]:
```suggestion
        $this->wa_id = $wa_id;
```

File: src/Message/Contact/Phone.php [hunk 3]
```diff
@@ -6,11 +6,14 @@ class Phone
 {
     protected string $number;
 
+    protected string $wa_id;
+
     protected PhoneType $type;
 
-    public function __construct(string $number, PhoneType $type)
+    public function __construct(string $number, PhoneType $type, string $wa_id = null)
```
Comment (netflie) [inline src/Message/Contact/Phone.php:13]:
```suggestion
    public function __construct(string $number, PhoneType $type, string $wa_id = '')
```

File: src/Request/RequestContactMessage.php [hunk 4]
```diff
@@ -31,10 +31,17 @@ protected function makeBody(): void
...
 
         foreach ($this->message->phones() as $phone) {
-            $this->body[$message_type][0]['phones'][] = [
+
+            $phone_array = [
                 'phone' => $phone->number(),
                 'type' => $phone->type()->getValue(),
             ];
+
+            if(!empty($phone->waId())) {
```
Comment (netflie) [inline src/Request/RequestContactMessage.php:40]:
```suggestion
            if (!empty($phone->waId())) {
```

File: src/Request/RequestContactMessage.php [hunk 5]
```diff
@@ -31,10 +31,17 @@ protected function makeBody(): void
         ];
 
         foreach ($this->message->phones() as $phone) {
-            $this->body[$message_type][0]['phones'][] = [
+
```
Comment (netflie) [inline src/Request/RequestContactMessage.php:34]:
```suggestion
```

**Review by netflie — APPROVED** [review]

**Pull request merged**